api4/file.go

```diff
@@ -725,3 +727,76 @@ func writeFileResponse(filename string, contentType string, contentSize int64, l
 
 	http.ServeContent(w, r, filename, lastModification, fileReader)
 }
+
+func searchFiles(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.Config().FeatureFlags.FilesSearch {
+		c.Err = model.NewAppError("searchFiles", "api.post.search_files.not_implemented.app_error", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	if !c.App.SessionHasPermissionToTeam(*c.App.Session(), c.Params.TeamId, model.PERMISSION_VIEW_TEAM) {
+		c.SetPermissionError(model.PERMISSION_VIEW_TEAM)
+		return
+	}
+
+	params, jsonErr := model.SearchParameterFromJson(r.Body)
+	if jsonErr != nil {
+		c.Err = model.NewAppError("searchFiles", "api.post.search_files.invalid_body.app_error", nil, jsonErr.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if params.Terms == nil || len(*params.Terms) == 0 {
+		c.SetInvalidParam("terms")
+		return
+	}
+	terms := *params.Terms
+
+	timeZoneOffset := 0
+	if params.TimeZoneOffset != nil {
+		timeZoneOffset = *params.TimeZoneOffset
+	}
+
+	isOrSearch := false
+	if params.IsOrSearch != nil {
+		isOrSearch = *params.IsOrSearch
+	}
+
+	page := 0
+	if params.Page != nil {
+		page = *params.Page
+	}
+
+	perPage := 60
+	if params.PerPage != nil {
```

Comment:
I think we should check for a max items per page here, something like:

```suggestion
	if params.PerPage != nil && *params.PerPage < 60 {
```

and if we make this change, I'd suggest to move that `60` to a `const`

Reply:
This is strictly based on what is done with posts, If you feel strong about this file a ticket to fix this in all the searches.